oclint-rules/include/oclint/AbstractASTVisitorRule.h

```diff
@@ -12,7 +11,41 @@ template<typename T>
 class AbstractASTVisitorRule : public AbstractASTRuleBase, protected clang::RecursiveASTVisitor<T>
 {
     friend class clang::RecursiveASTVisitor<T>;
+private:
+    virtual bool shouldTraverseFile(clang::SourceLocation startLocation)
+    {
+        clang::SourceManager *sourceManager = &_carrier->getSourceManager();
+
+        if (sourceManager->getMainFileID() == sourceManager->getFileID(startLocation))
```

Comment:
We had some crash report in the past where we need to check if the source manager's start location is a valid one, so could you please add that checking back?

Reply:
Fixed